From 228e49fabffa644ab7a6a03e98205f293115dc89 Mon Sep 17 00:00:00 2001
Message-Id: <228e49fabffa644ab7a6a03e98205f293115dc89.1445029944.git.jen@redhat.com>
From: John Snow <jsnow@redhat.com>
Date: Fri, 16 Oct 2015 16:26:24 -0500
Subject: [CHANGE] qtest/ide-test: disable flush-test
To: rhvirt-patches@redhat.com,
    jen@redhat.com

RH-Author: John Snow <jsnow@redhat.com>
Message-id: <1445012784-22604-2-git-send-email-jsnow@redhat.com>
Patchwork-id: 68169
O-Subject: [RHEL-7.2 qemu-kvm PATCH 1/1] qtest/ide-test: disable flush-test
Bugzilla: 1270341
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Wei Huang <wei@redhat.com>
RH-Acked-by: Jeff Nelson <jenelson@redhat.com>

One of the tests downstream causes a race that can result in build
failures. For 7.3, we intend to fix the test properly, but for 7.2,
we disable the test as a workaround.

Signed-off-by: John Snow <jsnow@redhat.com>
Signed-off-by: Jeff E. Nelson <jen@redhat.com>
---
 tests/ide-test.c | 38 --------------------------------------
 1 file changed, 38 deletions(-)

diff --git a/tests/ide-test.c b/tests/ide-test.c
index 51f9239..43b7fd6 100644
--- a/tests/ide-test.c
+++ b/tests/ide-test.c
@@ -425,43 +425,6 @@ static void test_identify(void)
     ide_test_quit();
 }
 
-static void test_flush(void)
-{
-    uint8_t data;
-
-    ide_test_start(
-        "-vnc none "
-        "-drive file=blkdebug::%s,if=ide,cache=writeback",
-        tmp_path);
-
-    /* Delay the completion of the flush request until we explicitly do it */
-    qmp("{'execute':'human-monitor-command', 'arguments': { "
-        "'command-line': 'qemu-io ide0-hd0 \"break flush_to_os A\"'} }");
-
-    /* FLUSH CACHE command on device 0*/
-    outb(IDE_BASE + reg_device, 0);
-    outb(IDE_BASE + reg_command, CMD_FLUSH_CACHE);
-
-    /* Check status while request is in flight*/
-    data = inb(IDE_BASE + reg_status);
-    assert_bit_set(data, BSY | DRDY);
-    assert_bit_clear(data, DF | ERR | DRQ);
-
-    /* Complete the command */
-    qmp("{'execute':'human-monitor-command', 'arguments': { "
-        "'command-line': 'qemu-io ide0-hd0 \"resume A\"'} }");
-
-    /* Check registers */
-    data = inb(IDE_BASE + reg_device);
-    g_assert_cmpint(data & DEV, ==, 0);
-
-    data = inb(IDE_BASE + reg_status);
-    assert_bit_set(data, DRDY);
-    assert_bit_clear(data, BSY | DF | ERR | DRQ);
-
-    ide_test_quit();
-}
-
 static void test_flush_nodev(void)
 {
     ide_test_start("");
@@ -505,7 +468,6 @@ int main(int argc, char **argv)
     qtest_add_func("/ide/bmdma/long_prdt", test_bmdma_long_prdt);
     qtest_add_func("/ide/bmdma/teardown", test_bmdma_teardown);
 
-    qtest_add_func("/ide/flush", test_flush);
     qtest_add_func("/ide/flush_nodev", test_flush_nodev);
 
     ret = g_test_run();
-- 
2.4.3

